components/suggest/src/store.rs

```diff
@@ -366,31 +366,37 @@ where
             return Ok(());
         };
 
-        let attachment: SuggestAttachment<T> =
-            serde_json::from_slice(&self.settings_client.get_attachment(&attachment.location)?)?;
-
-        writer.write(|dao| {
-            // Drop any suggestions that we previously ingested from
-            // this record's attachment. Suggestions don't have a
-            // stable identifier, and determining which suggestions in
-            // the attachment actually changed is more complicated than
-            // dropping and re-ingesting all of them.
-            dao.drop_suggestions(&record_id)?;
-
-            // Ingest (or re-ingest) all suggestions in the
-            // attachment.
-            ingestion_handler(dao, &record_id, attachment.suggestions())?;
-
-            // Remove this record's ID from the list of unparsable
-            // records, since we understand it now.
-            dao.drop_unparsable_record_id(&record_id)?;
-
-            // Advance the last fetch time, so that we can resume
-            // fetching after this record if we're interrupted.
-            dao.put_last_ingest_if_newer(record.last_modified)?;
-
-            Ok(())
-        })
+        let attachment_data = self.settings_client.get_attachment(&attachment.location)?;
```

Comment:
Should we also handle errors fetching the attachment in the same way?

Reply:
Ooh, good question! I was thinking no, because `get_attachment` just does the network request, and the error would most likely be a network error (or some other kind of error that we'd want to report). But an error when parsing means "we don't understand this attachment, and trying to fetch it again won't help".

WDYT?

Reply:
Yeah that makes sense to me as well.  I wasn't sure what work `get_attachment` was doing.